### ISSUE TITLE
Error when updating 

### ISSUE THREAD

**Issue by DarkClaw1** [xaya/xaya#75]
This is probably something dumb I did but it looks like my attempt to update to the latest branch failed:
```
xxx@xxx:~/xaya$ git pull origin master
remote: Enumerating objects: 2114, done.
remote: Counting objects: 100% (2114/2114), done.
remote: Compressing objects: 100% (253/253), done.
remote: Total 2460 (delta 1921), reused 2040 (delta 1860), pack-reused 346
Receiving objects: 100% (2460/2460), 1.12 MiB | 3.92 MiB/s, done.
Resolving deltas: 100% (2036/2036), completed with 635 local objects.
From https://github.com/xaya/xaya
 * branch                master     -> FETCH_HEAD
   f31f6a378..2c638d773  master     -> origin/master
Auto-merging test/functional/xaya_premine.py
Auto-merging test/functional/wallet_multiwallet.py
CONFLICT (content): Merge conflict in test/functional/wallet_multiwallet.py
Auto-merging test/functional/p2p_invalid_block.py
Auto-merging test/functional/auxpow_mining.py
CONFLICT (content): Merge conflict in test/functional/auxpow_mining.py
Auto-merging test/functional/auxpow_getwork.py
Removing src/wallet/test/accounting_tests.cpp
Auto-merging src/wallet/db.cpp
Auto-merging src/validation.cpp
Auto-merging src/util/time.h
Auto-merging src/util/time.cpp
Auto-merging src/util/system.h
Auto-merging src/util/system.cpp
Auto-merging src/util/strencodings.h
Auto-merging src/util/strencodings.cpp
Auto-merging src/util/moneystr.h
Auto-merging src/util/moneystr.cpp
Auto-merging src/qt/test/rpcnestedtests.cpp
Removing src/qt/callback.h
Auto-merging src/clientversion.cpp
Auto-merging src/chainparams.cpp
Auto-merging src/auxpow.cpp
Auto-merging share/examples/namecoin.conf
Removing doc/release-notes-pr13033.md
Removing doc/release-notes-pr12924.md
Removing doc/release-notes-pr12892.md
Removing doc/release-notes-pr12823.md
Removing doc/release-notes-pr10740.md
Removing doc/release-notes-pr10267.md
Removing depends/patches/qt/qfixed-coretext.patch
Removing depends/patches/qt/fix-cocoahelpers-macos.patch
Removing depends/packages/libSM.mk
Removing depends/packages/libICE.mk
Removing contrib/testgen/gen_base58_test_vectors.py
Removing contrib/rpm/bitcoin.te
Removing contrib/rpm/bitcoin.spec
Removing contrib/rpm/bitcoin.if
Removing contrib/rpm/bitcoin.fc
Removing contrib/rpm/README.md
Removing contrib/gitian-build.sh
Removing contrib/debian/watch
Removing contrib/debian/source/format
Removing contrib/debian/rules
Removing contrib/debian/patches/series
Removing contrib/debian/patches/README
Removing contrib/debian/gbp.conf
Removing contrib/debian/control
Removing contrib/debian/compat
Removing contrib/debian/changelog
Removing contrib/debian/bitcoind.manpages
Removing contrib/debian/bitcoind.lintian-overrides
Removing contrib/debian/bitcoind.install
Removing contrib/debian/bitcoind.examples
Removing contrib/debian/bitcoind.bash-completion
Removing contrib/debian/bitcoin-tx.manpages
Removing contrib/debian/bitcoin-tx.install
Removing contrib/debian/bitcoin-tx.bash-completion
Removing contrib/debian/bitcoin-qt.protocol
Removing contrib/debian/bitcoin-qt.manpages
Removing contrib/debian/bitcoin-qt.lintian-overrides
Removing contrib/debian/bitcoin-qt.install
Removing contrib/debian/bitcoin-qt.desktop
Removing contrib/debian/README.md
Auto-merging configure.ac
CONFLICT (content): Merge conflict in configure.ac
Removing build-aux/m4/ax_boost_program_options.m4
Automatic merge failed; fix conflicts and then commit the result.
xxx@xxx:~/xaya$ xayad
2018-12-08T19:36:46Z Xaya version v1.0.3.0-3827a2f80 (release build)
```
Now I see this:
```
xxx@xxx:~/xaya$ git pull origin master
error: Pulling is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
```
Sorry if this is just some misunderstanding of git...

**Comment by domob1812** [comment]
That looks to me like you had some changes in your git repository that are not compatible with the changes made by us in the latest version (in those files that are marked as conflict).

To recover, I think you can try something like `git merge --abort` and `git reset HARD origin/master`, which will reset your repository to the current version from us.  It may "delete" your changes if those were not made on a branch in your repository, though, so be careful.

**Comment by DarkClaw1** [comment]
That worked, I don't remember making any changes though. Maybe you can figure it out from this info:
```
xxx@xxx:~/xaya$ git show
commit 3827a2f805a8334ff5d298734b13e9f49e6ffc39 (HEAD -> 1.0, origin/1.0)
Author: Daniel Kraft <d@domob.eu>
Date:   Mon Sep 24 11:27:06 2018 +0200

    Bump version to 1.0.3.
    
    This will be the next version (if any is released) on the 1.0 branch,
    now that 1.0.2 has been tagged.

diff --git a/configure.ac b/configure.ac
index ef41805bf..46362b87a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@ dnl require autoconf 2.60 (AS_ECHO/AS_ECHO_N)
 AC_PREREQ([2.60])
 define(_CLIENT_VERSION_MAJOR, 1)
 define(_CLIENT_VERSION_MINOR, 0)
-define(_CLIENT_VERSION_REVISION, 2)
+define(_CLIENT_VERSION_REVISION, 3)
 define(_CLIENT_VERSION_BUILD, 0)
 define(_CLIENT_VERSION_IS_RELEASE, true)
 define(_COPYRIGHT_YEAR, 2018)
xxx@xxx:~/xaya$ git diff
diff --cc configure.ac
index 46362b87a,ccecbef02..000000000
--- a/configure.ac
+++ b/configure.ac
@@@ -1,10 -1,11 +1,16 @@@
  dnl require autoconf 2.60 (AS_ECHO/AS_ECHO_N)
  AC_PREREQ([2.60])
  define(_CLIENT_VERSION_MAJOR, 1)
++<<<<<<< HEAD
 +define(_CLIENT_VERSION_MINOR, 0)
 +define(_CLIENT_VERSION_REVISION, 3)
++=======
+ define(_CLIENT_VERSION_MINOR, 1)
+ define(_CLIENT_VERSION_REVISION, 99)
++>>>>>>> 2c638d7734b823c070dd66a17937c012c8e78c43
  define(_CLIENT_VERSION_BUILD, 0)
- define(_CLIENT_VERSION_IS_RELEASE, true)
+ define(_CLIENT_VERSION_RC, 0)
+ define(_CLIENT_VERSION_IS_RELEASE, false)
  define(_COPYRIGHT_YEAR, 2018)
  define(_COPYRIGHT_HOLDERS,[The %s developers])
  define(_COPYRIGHT_HOLDERS_SUBSTITUTION,[[Bitcoin Core, Namecoin and Xaya]])
diff --cc test/functional/auxpow_mining.py
index 79268e0a0,9229eedda..000000000
--- a/test/functional/auxpow_mining.py
+++ b/test/functional/auxpow_mining.py
@@@ -15,6 -19,8 +19,11 @@@ from test_framework.auxpow_testing impo
    getCoinbaseAddr,
    mineAuxpowBlockWithMethods,
  )
++<<<<<<< HEAD
++=======
+ 
+ from decimal import Decimal
++>>>>>>> 2c638d7734b823c070dd66a17937c012c8e78c43
  
  class AuxpowMiningTest (BitcoinTestFramework):
  
diff --cc test/functional/wallet_multiwallet.py
index ec8a2539c,8ab569a3c..000000000
--- a/test/functional/wallet_multiwallet.py
+++ b/test/functional/wallet_multiwallet.py
@@@ -35,6 -39,8 +39,11 @@@ class MultiWalletTest(BitcoinTestFramew
                  return wallet_dir(name, "wallet.dat")
              return wallet_dir(name)
  
++<<<<<<< HEAD
++=======
+         assert_equal(self.nodes[0].listwalletdir(), { 'wallets': [{ 'name': '' }] })
+ 
++>>>>>>> 2c638d7734b823c070dd66a17937c012c8e78c43
          # check wallet.dat is created
          self.stop_nodes()
          assert_equal(os.path.isfile(wallet_dir('wallet.dat')), True)
@@@ -269,6 -294,8 +297,11 @@@
          assert_equal(self.nodes[0].listwallets(), ['w1'])
          assert_equal(w1.getwalletinfo()['walletname'], 'w1')
  
++<<<<<<< HEAD
++=======
+         assert_equal(sorted(map(lambda w: w['name'], self.nodes[0].listwalletdir()['wallets'])), ['', os.path.join('sub', 'w5'), 'w', 'w1', 'w2', 'w3', 'w7', 'w7_symlink', 'w8', 'w8_copy', 'w9'])
+ 
++>>>>>>> 2c638d7734b823c070dd66a17937c012c8e78c43
          # Test backing up and restoring wallets
          self.log.info("Test wallet backup")
          self.restart_node(0, ['-nowallet'])
```
Anyway, if you want to assume this was just some random thing I messed up then I can close.

**Comment by domob1812** [comment]
In any case, this is related to the git workflow and not Xaya's code, so I'll close now.
